docs/proposals/tracing.md

```diff
@@ -0,0 +1,138 @@
+# Tracing
+
+## Context
+
+There are many tracing technologies in the market. To use tracing in your application you must instrument it with a library, therefore changing underlying technology is expensive (you have to change it in every app).
+That's why OpenTracing was created. Instrument your application once and change the backend (tracer) when you want. Around the same time OpenCensus was created which had a same goal.
+OpenCensus and OpenTracing have merged into CNCF OpenTelemetry, but the work is still in progress.
+
+Envoy has a [built-in support](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/observability/tracing) for tracing and we want to leverage this support in Kuma.
+
+### Supported backends
+
+Envoy supports following backends: Lightstep, Zipkin,  OpenTracing (with dynamically loaded library), Datadog, OpenCensus (including GCP Stackdriver) and gRPC service for custom implementation.
+
+### Tracing as a Bootstrap Configuration
+
+You can define only one tracing backend for an Envoy and this setting is a part of the bootstrap configuration (the one that CP delivers before the start of DP).
+That means that if you want to change it, you have to restart Envoy.
```

Comment:
@subnetmarco We could contribute to upstream Envoy to make tracing configuration dynamic.

Is it important for `Kuma` to have tracing configuration completely dynamic ?